src/components/BulkEnrollmentModal/index.jsx

```diff
@@ -53,24 +57,60 @@ class BulkEnrollmentModal extends React.Component {
     onClose();
   }
 
+  validateFormData(formData) {
+    const userEmailsKey = EMAIL_ADDRESS_TEXT_FORM_DATA;
+    const emailsCSVKey = EMAIL_ADDRESS_CSV_FORM_DATA;
+
+    /* eslint-disable no-underscore-dangle */
+    // The 'subject' field is not required here.
+    let errors = {
+      _error: [],
+    };
+
+    if (!formData[userEmailsKey] && !formData[emailsCSVKey]) {
+      const message = 'Either user emails or emails csv must be provided.';
+      errors[userEmailsKey] = message;
+      errors[emailsCSVKey] = message;
+      errors._error.push(message);
+    }
+
+    const emailFieldErrors = validateEmailAddressesFields(formData);
+    errors = mergeErrors(errors, emailFieldErrors);
```

Comment:
Missing fields? Not sure I follow, but as an example that may illustrate what you're getting at here:
You could upload a CSV with invalid email addresses (think not having a domain part of the email, ie just `bobb32` instead of `bobb32@gmail.com`) and you could add individual emails that are also not valid and both will trigger validation errors that will be displayed at the top of the modal.

Reply:
okay to address this area before I close it:
1. We wouldn't have both the message saying you're missing values from both fields AND errors from validating emails.
2. However, we can remove the mergeErrors call because as Alex pointed out, with the logic as is we'd only have one or the other and as far as keeping formatting consistent, validateEmailAddressesFields always returns a dict so we can just reuse that.